### ISSUE TITLE
Bad Chord name display from Sibelius XML - m7b5

### ISSUE THREAD

**Issue by AlbertHart** [opensheetmusicdisplay/opensheetmusicdisplay#590]
Here are two m7b5 chords in Sibelius ang Musescore displayed in the OSMD Demo

![image](https://user-images.githubusercontent.com/8965352/67168575-259f0080-f363-11e9-897e-4a5758ee67c4.png)

Here is the <harmony> entries from each export:

Sibelius Harmony Entry

```
<harmony print-frame="no">
        <root>
          <root-step>F</root-step>
          <root-alter>1</root-alter>
          </root>
        <kind text="m7b5">half-diminished</kind>
        </harmony>
```

MyaseScore Harmony Entry

```
<harmony color="#000000" default-y="25">
    <root>
     <root-step>F</root-step>
     <root-alter>1</root-alter>
    </root>
    <kind>minor-seventh</kind>
    <degree>
     <degree-value>5</degree-value>
     <degree-alter>-1</degree-alter>
     <degree-type>alter</degree-type>
    </degree>
    <staff>1</staff>
   </harmony>
```

After loading the score, osmd contains:

measure.staffEntries[0].sourceStaffEntry.chordSymbolContainer.chordKind = 6 for the Sibelius chord

and

measure.staffEntries[0].sourceStaffEntry.chordSymbolContainer.chordKind = 9 for the MuseScore Chord.
"

The functionl: getTextFromChordKindEnum() looks like it should work OK for 

ChordKind 9 (halfdiminished), but I suspect the translation to VexFlow is passing the wrong chord.

case ChordSymbolEnum.halfdiminished:
                text += "m7b5";
                break;

I have attached the two music HTML files.

[Am7b5.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/3748576/Am7b5.zip)

**Comment by AlbertHart** [comment]
If I remove the degree alter:

```
<degree>
     <degree-value>5</degree-value>
     <degree-alter>-1</degree-alter>
     <degree-type>alter</degree-type>
    </degree>
```
It works OK.

degree-alter is supposed to alter the degree of the chord - not alter the chord itself.

'The degree-alter type represents the chromatic alteration for the current degree."

But I suspect the degree-type alter is being read into the ChordDegreeText.alter value.

```
switch (chordSymbol.ChordDegree.text) {
                case ChordDegreeText.add:
                    text += "add";
                    break;
                case ChordDegreeText.alter:
                    text += "alt";
                    break;
                case ChordDegreeText.subtract:
                    text += "sub";
                    break;
                default:
            }
```

I am not set up to debur or compile at this level, but if you are, perhaps you can trace this down.

**Comment by AlbertHart** [comment]
I found a work around for this. 

If I enter the chord into Sibelius as Fhalfdim, it displays with a half-dim symbol in Sibelius, but displays properly in OSMD

![image](https://user-images.githubusercontent.com/8965352/67528019-9b61df80-f675-11e9-9cf4-d97c5b9c7d09.png)

**Comment by MilesMorel** [comment]
I also found that using the default font it changed “alt” or “b9” to “add9” and “maj7 (#11)” to "maj7add11"

**Comment by MilesMorel** [comment]
Also, if you copy/paste chord symbols into Sibelius, it renders flats and minor/major symbols incorrectly. If it can't handle certain fonts I wonder if there's some kind of linting that can be done on xml with chord symbols or if it can replace those fonts with basic fonts to attempt to render? In any case, getting a full dictionary of chords seems like it should fairly easy. I've seen the ChordSymbolEnum but guessin that these actually live in Vexflow and not OSMD?

**Comment by sschmidTU** [comment]
ChordSymbolEnum 'lives' in OSMD.

> I also found that using the default font it changed “alt” or “b9” to “add9” and “maj7 (#11)” to "maj7add11"

In OSMD or Sibelius? If OSMD displays "add", that's because there's "add" in the `degree-type` in the XML.

We could offer developers customizing what OSMD displays here similar to:
```typescript
EngravingRules.setChordSymbolLabelText(key: ChordSymbolEnum, value: string)
```
in `ChordSymbolContainer.ts:calculateChordText()`.

**Comment by MilesMorel** [comment]
Thank you @sschmidTU - I'm going to try to get some more info together and post a specific example later tonight.

**Comment by sschmidTU** [comment]
@MilesMorel see https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/930#issuecomment-759057797, this may be relevant to the solution as well. maybe we should just use the chord-symbol library mentioned there.

**Comment by MilesMorel** [comment]
I agree. Sounds like `chord-symbol` would be a nice option - cool library!

**Comment by sschmidTU** [comment]
Mostly fixed in #933.

![image](https://user-images.githubusercontent.com/33069673/105732355-a4082c00-5f30-11eb-8722-f2443f722a7a.png)

If you want it to display "F#m7b5" instead of "F#m7(alt b5)", use this:
`osmd.EngravingRules.addChordName("m7b5", "minorseventh", [], ["b5"], []);`
![image](https://user-images.githubusercontent.com/33069673/105732136-6d321600-5f30-11eb-95c6-e53001b8c347.png)

There's also this new function in EngravingRules:
```typescript
public renameChord(altName: string, newAltName: string)
```
